internal/util/util.go

```diff
@@ -22,3 +31,45 @@ func LookupImage(ctx *types.SystemContext, store storage.Store, image string) (*
 	}
 	return localImage, nil
 }
+
+// ExportFromReader reads bytes from given reader and exports to external tar, directory or stdout.
+func ExportFromReader(input io.Reader, opts define.BuildOutputOption) error {
+	var err error
+	if !filepath.IsAbs(opts.Path) {
+		opts.Path, err = filepath.Abs(opts.Path)
+		if err != nil {
+			return err
+		}
+	}
+	if opts.IsDir {
+		// preserve ownership when invoked as root
+		noLChown := false
+		if unshare.IsRootless() {
+			noLChown = true
+		}
+
+		err = os.MkdirAll(opts.Path, 0700)
+		if err != nil {
+			return err
```

Comment:
`errors.Wrapf()` this, so that if we fail here, the caller can get a stack trace that points to this line.